docs/NXP_PXP_readme.md

```diff
@@ -0,0 +1,59 @@
+
```

Comment:
Can we move the README to the [docs](https://docs.lvgl.io/latest/en/html/get-started/nxp.html) instead?  If so a few lines of introduction would be great about getting started, recommended dev boards, support in MCUXpresso etc.

Reply:
Hello Gabor, 
what do you mean by moving readme to docs? Are you suggesting moving toplevel `README.md` to `docs` folder? Or are you suggesting that we should merge the content of `NXP_PXP_readme.md` into toplevel `REDME.md`? I'll add more information in meantime.

Reply:
I meant to move it to https://docs.lvgl.io/latest/en/html/get-started/nxp.html
I'm sure it will get more visibility there.

Reply:
I see, sure, that's great idea. Let's wait a little bit, I'll add more information as you suggested above, so it will be more explanatory for new users.

Reply:
Hi, 
could we reuse general description for get-started page from https://lvgl.io/developers#nxp and keep readme in docs? The thing is that this readme is related **only** to PXP device configuration, but not all supported NXP platforms have this PXP.  So for general info about NXP platforms and how to get it running, user can browse information available via red "Get started" button.

Reply:
Sure we can use that text. It was written by NXP anyway. :slightly_smiling_face: 

I'm ok with describing mainly PXP in this README but if move it to the [docs](https://docs.lvgl.io/latest/en/html/get-started/nxp.html) there should be at least one paragraph to generally describe that "LVGL works with MCUXPresso and for more info click here." Later this paragraph can be extended with a step-by-step guide to getting started.


Just to confirm:
- move[ this README](https://github.com/lvgl/lvgl/pull/1702/files#diff-e35286a996b1b8660fc1184dc8fe6dcc) to [LVGL docs](https://docs.lvgl.io/latest/en/html/get-started/nxp.html)
- add a general introduction paragraph
- keep the "Get started" button on https://lvgl.io/developers#nxp to point to nxp.com.

Do you agree?

Reply:
My point was to reuse(=copy) that general introducion paragraph from https://lvgl.io/developers#nxp into https://docs.lvgl.io/latest/en/html/get-started/nxp.html - meaning that PXP details from [docs/NXP_PXP_readme.md](https://github.com/lvgl/lvgl/blob/a01b7a448bd1a554d70e0c278e47e7c1c60962c6/docs/NXP_PXP_readme.md) will not appear there at all. It's specific for few controllers only and when user download SDK for MCU with PXP, it will be enabled and all integration work is already done. It may be better to maintain list of LVGL enabled boards on NXP page, so we don't have sync-problems when new and new MCUs are enabled by our teams. Therefore I thought that https://docs.lvgl.io/latest/en/html/get-started/nxp.html is general and should not contain deteails specific only for few MCUs.

Regarding your question about commits into our forks - unfortunatelly this will not be possible. So either I will do the change as we discuss, or you can do the changes in main repo when PR is merged. Would it be okay like this?

Reply:
> My point was to reuse(=copy) that general introducion paragraph from https://lvgl.io/developers#nxp into https://docs.lvgl.io/latest/en/html/get-started/nxp.html 

It's ok with some links to NXP pages.

However, we should keep the docs in one place. It's not a problem if PXP docs is related to only a few MCUs but once we have docs for it it should be in an easy to find a place. (I.e. among all the other docs)

>  It may be better to maintain list of LVGL enabled boards on NXP page, so we don't have sync-problems when new and new MCUs are enabled by our teams. 

I agree to not list the supported MCU in LVGL's side just add a link to an NXP page. 

> Regarding your question about commits into our forks - unfortunatelly this will not be possible. So either I will do the change as we discuss, or you can do the changes in main repo when PR is merged. Would it be okay like this?

Sure, I just wondered why it wasn't working in the usual way.

Reply:
I see. There's no offline doc in the repo. How could we reference web-based documentation (from MCU SDK), if we move document to web-only? On [doc page](https://docs.lvgl.io/dev/en/html/) there's listbox with latest, development, v7 versions. For V7 there's only latest one - v7.3.1. How can we refer to web doc for projects based on older release e.g. 7.1?

Reply:
The older versions should be [tagged](https://github.com/lvgl/docs/tags) in the docs repo. Some of them are missing, but I think that's because the release script was having issues.

Reply:
But this is not browsable online, it's just ZIP file? It may be good to add all releases to the selection on web so even older doc is available online on https://docs.lvgl.io/latest/en/html/. 

Now I understand Gabor's idea. I'll add general paragraph, make PXP content as separate chapter, and create PR to docs repository.

Reply:
In my opinion, there's little to no reason to not use the latest minor version. There are no significant breaking API changes between 7.0 and 7.3. Do you think there's a compelling reason to use 7.1, specifically?

Reply:
> Some of them are missing, but I think that's because the release script was having issues.

Oh yes... Some issues show up only when I make the release and can't test them earlier. :slightly_frowning_face: 
I hope the script will stabilize soon.

>  In my opinion, there's little to no reason to not use the latest minor version. There are no significant breaking API changes between 7.0 and 7.3.

I agree.  The user should rather update at least to the latest `release/vX`. (`latest` can be other a major version ahed so it's not sure he wants to update to it.).

Besides we have releases every 2 weeks so the list would be very long very soon. 

> Do you think there's a compelling reason to use 7.1, specifically?

Ideas and feedback are always welcome, so I'm curious if you see a use-case for it.

Reply:
Idea of keeping all docs versions is just for consistency. It comes from the topology, as code base is separated from docs. There is no guarantee that users with complex project will keep LVGL always up to date. Once they have stable environment, every update can bring new issues, new testing etc. So it's quite possible that they will stay at e.g. 7.1, even thouh there's already 7.2, 7.3, 7.4 etc., unless they see strong reason for update. And when they open the doc page, they don't find 7.1. There's 7.4. and 7.5-dev only. One could check code differences between 7.1 and 7.5-dev to see whether doc is still valid or not. Also, one should keep the snapshot of docs repo in their project, but who knows :-) Working with 7.1 code and looking at 7.4 doc is little bit risky. 

Btw. what is the reason for separate docs repository? Is there any benefit not having it all in one repo?

Reply:
I see your point. Theoretically, people can find the version of the related docs with tags: https://github.com/lvgl/docs/tags

However, it's really not that user friendly.

> Btw. what is the reason for separate docs repository? Is there any benefit not having it all in one repo?

I see two reasons:
1. The docs repo is huge (~1GB). With this LVGL would be very large. 
2. Normally users don't need to docs as plain text but in HTML or PDF. So why keep the source of the docs on the core repo?

However, the HTML pages (+images) are only 20 MB. So we can them on every release.
What do you think about it?

Reply:
I'd suggest that we use a GitHub release to attach a PDF file to each tag in the docs repository. It seems like the most user-friendly solution without needing to copy information from one repo to the other.

Reply:
Sounds good, but we should attach the HTML too, shouldn't we?

Reply:
That would also work. My main reason for suggesting a PDF is that it is more self-contained than many HTML files.

Reply:
I see. But once space is free I see no reason to skip the website.